lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift

```diff
@@ -146,8 +137,37 @@ extension SourceManager.MacroExpansionContext: MacroExpansionContext {
     let offsetWithinSyntaxNode =
       rawPosition.utf8Offset - node.position.utf8Offset
 
+    var location = exportedSourceFile.sourceLocationConverter.location(
+      for: rootPosition.advanced(by: offsetWithinSyntaxNode)
+    )
+
+    switch filePathMode {
+    case .fileID:
+      // The `SourceLocationConverter` in `exportedSourceFile` uses `filePath` as the file mode. When the `fileID` mode
+      // is requested, we need to adjust the file and presumed file to the `fileID`.
+      let fileID = "\(exportedSourceFile.moduleName)/\(exportedSourceFile.fileName.basename)"
+      var adjustedFile = location.file
+      if adjustedFile == exportedSourceFile.fileName {
+        adjustedFile = fileID
```

Comment:
What does it mean if they aren't equal? I thought that would be the presumed file case, but apparently not?